pkg/controller/direct/cloudbuild/workerpool_controller.go

```diff
@@ -109,7 +109,7 @@ func (m *model) AdapterForObject(ctx context.Context, reader client.Reader, u *u
 			return nil, err
 
 		}
-		obj.Spec.PrivatePoolConfig.NetworkConfig.PeeredNetworkRef.External = networkRef.String()
+		obj.Spec.PrivatePoolConfig.NetworkConfig.PeeredNetworkRef.External = networkRef.Id()
```

Comment:
Maybe we should name this form `ExternalRef()`?

Reply:
Do you mean the proposal Yuwen shared [here](https://docs.google.com/document/d/1A6YSEygeVJuXefb9cH52RiqEqp15nI5-UHNqG1T08kg/edit?tab=t.0#heading=h.yro65jsl04g3)? I'll keep that in mind, but I think that's a more generic change, we do not support status.ExternalRef for now. We can chat about that and address in a future PR.